src/java/org/apache/cassandra/cache/ChunkCache.java

```diff
@@ -125,6 +131,39 @@ public boolean equals(Object obj)
         }
     }
 
+    static class BufferFuture
+    {
+        private final Key key;
+        private final Buffer buffer;
+        private boolean loaded;
+
+        public BufferFuture(Key key, Buffer buffer)
```

Comment:
Nit: The `buffer` seems to be dependent on the `key` (at least this is how it is obtained). 
I feel the data flow would be simpler if the allocation of the buffer was moved to this constructor, and instead of taking a reference to a Buffer, it just took the reference to the `BufferPool` and the `Key`.